### PR TITLE
Image paste for Chrome

### PR DESCRIPTION
**Pull request by tiddydl**
Allow image paste, for the moment only working on Chrome (not checked on IE)

### CODE REVIEW

File: src/js/images.js [hunk 1]
```diff
@@ -113,14 +114,45 @@
...
+     * @param {Blob} blob with image (optional)
+     *
      * @return {void}
      */
 
-    Images.prototype.add = function () {
+    Images.prototype.add = function (blob) {
         var that = this,
             $file = $(this.templates['src/js/templates/images-fileupload.hbs']());
+        blob = (typeof blob === "undefined") ? 0 : blob;
```
Comment (j0k3r) [inline src/js/images.js]: You forgot one indentation.

**Comment by j0k3r** [comment]
Do you know if it will be hard to handle for Firefox ?

**Comment by tiddydl** [comment]
with this way it only works in Chrome because the Clipboard items in other browsers expires before arriving to the paste event... I'm going to try in other ways!

**Comment by orthes** [comment]
Thank you. I will look at this later today.

**Comment by orthes** [comment]
Closes #87

**Comment by tiddydl** [comment]
The way for making it cross-browsing is to paste the image into a contenteditable div (works in all browsers except Chrome, but for Chrome we have the other way), and translate the base64 image data into blob. 
The problem is that the medium plugin crashes the paste event into that div, so the div has to be created outside the editor (child of body). That it already works.

So, the idea is to show and focus that div when the active <p> dom get focus. When we paste into that div, if it contains an image, we execute the paste event > blob > add to jquery-file-upload > voilà. If not (or if we start typing), we hide the div and we return focus to original <p> dom.

Problem: maybe is too much for that feature? So I propose to make this editable div at core-level (show it only if addons have initiated), and use it also to auto-embed things if we paste any embed URLs, without having to click on embed button. What do you think?

**Comment by orthes** [comment]
I've finally got some time to look into you PR, but it doesn't work for me. I don't know what I'm missing. I'm testing it in Google Chrome 40 on Mac OS X.

When I copy and paste an image, it pastes only a plain text file name. Looking into the code it uses [fileupload's add function](https://github.com/tiddydl/medium-editor-insert-plugin/commit/7237b70943e9c23ed2a1d5626982f42620e3a20f#diff-2627f4673527bccc425ad1f8f9e933b7R177) and thus addon's uploadAdd function. But uploadAdd function [relies on existing .medium-insert-active element](https://github.com/orthes/medium-editor-insert-plugin/blob/1.0/src/js/images.js#L156) which is not created when using your paste function.

What am I missing? Does it work for you?

**Comment by tiddydl** [comment]
The matter is that you have to copy directly the image (from a website, from Mac image preview, from photoshop...), not the file. I was looking for copying the file, but it doesn't work, if only works with drag&drop.
Now I have a version that copy-pastes in all browsers, maybe I can profite some functions to extend to file drag&drop functionality.

**Comment by orthes** [comment]
That would be great.

**Comment by Dmatafonov** [comment]
This is how it works at our site, might help:

```javascript
//on paste event

if (e.clipboardData) {
            var items = e.clipboardData.items;

            if (items) {
                for (var i = 0; i < items.length; i++) {
                    if (items[i].type.indexOf("image") !== -1) {
                        var blob = items[i].getAsFile();

                        // Send file to server
                        var reader = new FileReader();
                        reader.onload = function () {
                                   //make loading image and actually send file as base64 to server
                        };
                        reader.readAsDataURL(blob);
                    }
                }
            }
        }

```

**Pull request closed without merging**